clang/lib/3C/CastPlacement.cpp

```diff
@@ -181,17 +185,26 @@ CastPlacementVisitor::getCastString(ConstraintVariable *Dst,
         Suffix = ", " + Bounds + ")";
       }
     }
-    return std::make_pair("_Assume_bounds_cast<" +
-                              Dst->mkString(Info.getConstraints(), false) +
-                              ">(",
-                          Suffix);
+    // The destination's type may be generic, which would have an out-of-scope
+    // type var, so use the already analysed local type var instead
+    std::string Type;
+    if (TypeVar != nullptr) {
+      Type = "_Ptr<" +
+          TypeVar->mkString(Info.getConstraints(),
+                               false, false, true) +
```

Comment:
iirc the `true` at the end of the argument list is for `EmitPointee`, so instead of wrapping this in `_Ptr` after passing that option, can you just stick with `EmitPointee=false` and not manually wrap it in `_Ptr`?

Reply:
The problem with that is that the pointer is wild, so the mkString call ends up as a `int*`. The cast is assuming a checked type, so we need to force it.

Reply:
In that case, pointers to pointers are still a problem:
```c
  int **mk_ptr(void);
  _For_any(T) void free_ptr(_Ptr<T> p_ptr) {}
  void work (void) {
    int **node = mk_ptr();
    free_ptr<_Ptr<int>>(node);
  }
```

Reply:
The pointer is wild, but the user has incorrectly instantiated `free_ptr` with a safe type. 3C doesn't rewrite the user-supplied instantiation, so there's a disconnect that doesn't compile. That seems like a good result, except that we want to guarantee compilation in this case (non -alltypes). What would you expect as a result?

Reply:
also the example doesn't compile to begin with, for similar problem.